### ISSUE TITLE
"OSError: [Errno 25] Inappropriate ioctl for device" when running SB Manager Tests in GitHub Actions

### ISSUE THREAD

**Issue by mdmintz** [seleniumbase/SeleniumBase#1549]
### "OSError: [Errno 25] Inappropriate ioctl for device" when running SB Manager Tests in GitHub Actions

```python
terminal_width = os.get_terminal_size()[0]

OSError: [Errno 25] Inappropriate ioctl for device
```

This can be fixed by changing how the terminal width is calculated.

A few possibilities, depending on the OS and Python version:

```python
import os
terminal_width = os.get_terminal_size().columns

import subprocess
terminal_width = int(subprocess.check_output(['stty', 'size']).split()[1])

import shutil
terminal_width = shutil.get_terminal_size((80, 20)).columns
```

**Comment by mdmintz** [comment]
Note: ``stty: stdin isn't a terminal`` occurred on a Mac when trying:

```python
import subprocess
terminal_width = int(subprocess.check_output(['stty', 'size']).split()[1])
```

Probably due to ``pytest -n 4`` multiprocess issues where the terminal captured output.

I'll have to consider that when trying a new solution.

**Comment by mdmintz** [comment]
This seems to work on MacOS / Linux:

```python
import subprocess
int(subprocess.check_output(['tput', 'cols']))
```

**Comment by mdmintz** [comment]
I rushed together a solution, and it works well enough for what I currently need:

```python
import os
import shutil
import subprocess
import sys

def is_windows():
    platform = sys.platform
    if "win32" in platform or "win64" in platform or "x64" in platform:
        return True
    else:
        return False

def get_terminal_width():
    width = 80  # default
    try:
        width = os.get_terminal_size().columns
    except Exception:
        try:
            if is_windows():
                raise Exception("Don't even try 'tput cols' on Windows!")
            width = int(subprocess.check_output(["tput", "cols"]))
        except Exception:
            try:
                width = shutil.get_terminal_size((80, 20)).columns
            except Exception:
                pass
    return width
```

**Comment by mdmintz** [comment]
Released in ``4.6.1`` -  https://github.com/seleniumbase/SeleniumBase/releases/tag/v4.6.1
